pkg/filters/args.go

```diff
@@ -92,13 +95,33 @@ func (filter *ArgFilter) Parse(filterName string, operatorAndValues string, even
 		return InvalidEventArgument(argName)
 	}
 
+	var err error
+	if argName == "syscall" {
+		err = filter.parseSyscallFilter(id, operatorAndValues)
+	} else {
+		err = filter.parseFilter(id, argName, operatorAndValues, func() Filter {
+			return NewStringFilter()
```

Comment:
Do you expect to have other types here ? is that the reason why you want a constructor pointer ?

Reply:
In the future when we better map out argument types (i'm prototyping that in my spare time), we can easily map an argument to a filter constructor.